serving/src/main/java/feast/serving/configuration/LettuceConfig.java

```diff
@@ -0,0 +1,41 @@
+package feast.serving.configuration;
+
+import feast.core.StoreProto;
+import feast.serving.specs.CachedSpecService;
+import io.lettuce.core.RedisClient;
+import io.lettuce.core.RedisURI;
+import io.lettuce.core.api.StatefulRedisConnection;
+import io.lettuce.core.codec.ByteArrayCodec;
+import io.lettuce.core.resource.ClientResources;
+import io.lettuce.core.resource.DefaultClientResources;
+import org.springframework.beans.factory.ObjectProvider;
+import org.springframework.context.annotation.*;
+
+@Configuration
+public class LettuceConfig {
```

Comment:
I have mixed feelings about this implementation. `LettuceConfig` here is a top level configuration for an optional online store. We already have Cassandra and Redis as store options, and I think we will soon have support for more types. Even if we limit it to 1 online store and 1 batch store, ideally we wouldn't have an inconsistency between how these clients are instantiated. 

What do you think about
1. Reusing another configuration class like ServingService to create the beans.
2. Creating a new configuration class that isn't tied to just Redis/Lettuce

Reply:
I have made redisclient as optional by checking if the host is set in redisconfig. `RedisClient` and `conenction` will be null and not initialized. We can do it more explicitly by checking `specService.getStore().getType()` and return null redisConf? 

For other configurations also we can check the store type and initialize. Does it make sense?

Reply:
Yes, I think we can put all store types in a configuration and pass a bean into while creating ServingService(). Connection Beans will be initialised based on the store type.
```
@Bean 
ServingService servingService(..., StorageConfiguration storeConf) {
....
case REDIS:
 servingService = new RedisServingService(storeConf.getRedisConnection(), specService, tracer);
CASE REDISCLUSTER:
 servingService = new RedisServingService(storeConf.getRedisClusterConnection(), specService, tracer);
}
```
and 
```
@Configuration
class StorageConfiguration {
RedisConnection connection;
RedisClusterConnection clusterConnection; 
...
RedisConnection getRedisConnection() {return this.connection;}
RedisClusterConnection getRedisClusterConnection{return this.clusterConnection;}
....
}
```